### ISSUE TITLE
Window client size is incorrect due not using AdjustWindowRect

### ISSUE THREAD

**Issue by silbinarywolf** [gonutz/d3d9#11]
**The problem**

When you create a window using the Win32 API, it assumes you're giving the width / height including border sizes. This means when you create a window, it ends up being off by a few pixels or so.

![image](https://user-images.githubusercontent.com/3859574/127734261-8e3fa31b-4bc0-4d19-bf3b-0fec49fe5924.png)
The above is a screenshot of the static_triangle sample. The size here is: 624x441 not the expected 640x480.

**Found the solution via this stack overflow answer:** 
- https://stackoverflow.com/questions/4843806/winapi-create-a-window-with-a-specified-client-area-size
- https://stackoverflow.com/a/25879124/5013410

**The solution**

Examples should be updated to make this API call so that the client window size is correct.
If we do that, then the window will be correct.

![image](https://user-images.githubusercontent.com/3859574/127735383-e6054824-b587-46fa-8fae-1a5bbda4032c.png)
_Size of this image is 640x480_

**Comment by gonutz** [comment]
You are right, the borders are not taken into account by the 640x480 given to CreateWindow. In this case though, that was actually my intent. I have tried this library on virtual Windows XP machines with 640x480 resolution. In this setup the window should not be larger than the screen, I actually really mean a 640x480 window, not client here. For this basic example the thing to show is how to get a triangle up on the screen. It is not intended to show inane Windows API quirks. They annoy me, I know about them, but this is not the place for them I think.

But thanks for the suggestion. I guess a way to solve this is to have multiple tutorials/examples, going from small-scale to large. We could have a nice tutorial with all these real-world problems that someone will face trying to create a real desktop 3D application, concentrating on one topic at a time. Unfortunately this is very time consuming and the thing I do not have right now is time...
